src/ServiceControl/SagaAudit/SagaUpdatedHandler.cs

```diff
@@ -1,24 +1,102 @@
 ﻿namespace ServiceControl.SagaAudit
 {
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading;
     using System.Threading.Tasks;
     using EndpointPlugin.Messages.SagaState;
+    using Newtonsoft.Json.Linq;
     using NServiceBus;
-    using ServiceControl.Persistence;
+    using ServiceControl.Connection;
+    using ServiceControl.Infrastructure.DomainEvents;
 
     class SagaUpdatedHandler : IHandleMessages<SagaUpdatedMessage>
     {
-        public SagaUpdatedHandler(ISagaAuditDataStore store)
+        public SagaUpdatedHandler(IDomainEvents domainEvents, IPlatformConnectionBuilder connectionBuilder)
         {
-            this.store = store;
+            this.domainEvents = domainEvents;
+            this.connectionBuilder = connectionBuilder;
         }
 
-        public Task Handle(SagaUpdatedMessage message, IMessageHandlerContext context)
+        public async Task Handle(SagaUpdatedMessage message, IMessageHandlerContext context)
         {
-            var sagaSnapshot = SagaSnapshotFactory.Create(message);
+            await WarnOncePerEnpointPerDay(message);
 
-            return store.StoreSnapshot(sagaSnapshot);
+            if (auditQueueName is null || nextAuditQueueNameRefresh < DateTime.UtcNow)
+            {
+                await RefreshAuditQueue();
+            }
+
+            if (auditQueueName is null)
+            {
+                throw new Exception("Could not determine audit queue name to forward saga update message. The ServiceControl remote audit instance ");
+            }
+
+            await context.ForwardCurrentMessageTo(auditQueueName);
+        }
+
+        async Task RefreshAuditQueue()
+        {
+            await semaphore.WaitAsync();
+            try
+            {
+                if (auditQueueName != null && nextAuditQueueNameRefresh > DateTime.UtcNow)
+                {
+                    return;
+                }
+
+                var connectionDetails = await connectionBuilder.BuildPlatformConnection();
+
+                if (connectionDetails.ToDictionary().TryGetValue("SagaAudit", out var sagaAuditObj) && sagaAuditObj is JObject sagaAudit)
+                {
+                    auditQueueName = sagaAudit["SagaAuditQueue"].Value<string>();
+                    nextAuditQueueNameRefresh = DateTime.UtcNow.AddMinutes(5);
+                }
+            }
+            catch (Exception)
+            {
```

Comment:
This could happen also when we cannot connect to the remote instance that is handling SagaAudit messages. 

In such a case, in order to prevent a message loss scenario I think we should consider:
- Either throwing an exception and making sure SagaAudit messages surface in the FailedMessages view in ServiceControl (when could add a custom recoverability policy to make sure that AuditMessages are moved to the error queue on the first exception)
- or store the messages that cannot be forwarded and tell the users to export/import the `SagaSnapshot` collection by hand.

I prefer the first option.

Reply:
Agreed, just throw here so that recoverability can deal with it